R/pkg/R/column.R

```diff
@@ -203,3 +203,49 @@ setMethod("%in%",
             jc <- callJMethod(x@jc, "in", table)
             return(column(jc))
           })
+
+
+#' when
+#'
+#' Evaluates a list of conditions and returns one of multiple possible result expressions.
+#' If otherwise is not defined at the end, null is returned for unmatched conditions.
+#'
+#' @rdname column
+setMethod("when",
+          signature(x = "Column", y = "character", z = "ANY"),
+          function(x, y, z) {
+            condition <- y
+            value <- z
+            jc <- callJMethod(x@jc, "when", condition, value)
+            column(jc)
+          })
+
+#' otherwise
+#' 
+#' Evaluates a list of conditions and returns one of multiple possible result expressions.
+#' If otherwise is not defined at the end, null is returned for unmatched conditions.
+#'
+#' @rdname column
+setMethod("otherwise",
+          signature(x = "Column", value = "ANY"),
+          function(x, value) {
+            value <- ifelse(class(value) == "Column", value@jc, value)
+            jc <- callJMethod(x@jc, "otherwise", value)
+            column(jc)
+          })
+
+#' otherwise
+#'
+#' This is an alias of `otherwise` as an infix operator.
+#'
+#' @rdname column
+#' @aliases otherwise
+#' @examples
+#' \dontrun{
+#'   when(df$x > 1 & df$y < 10, 0) %otherwise% 1
+#' }
+setMethod("%otherwise%",
```

Comment:
Alright. I'll remove `%otherwise%`. We could also implement `ifelse` with a combination of `when` and `otherwise` like below. Or should we implement `ifelse` as a native expression function in Scala?

```
setGeneric("ifelse", function(condition, e1, e2) { standardGeneric("ifelse") })
setMethod("ifelse", signature(condition = "Column", e1 = "ANY", e2 = "ANY"),
  function(condition, e1, e2) {
    e1 <- ifelse(class(e1) == "Column", e1@jc, e1)
    e2 <- ifelse(class(e1) == "Column", e2@jc, e2)
    jc <- callJStatic("org.apache.spark.sql.functions", "when", condition, e1)
    jc <- otherwise(jc, e2)
    column(jc)
  }
)
```